packages/core/src/routes/resource.ts

```diff
@@ -0,0 +1,106 @@
+import { Resources } from '@logto/schemas';
+import { object, string } from 'zod';
+
+import koaGuard from '@/middleware/koa-guard';
+import koaPagination from '@/middleware/koa-pagination';
+import {
+  findTotalNumberOfResources,
+  findAllResources,
+  findResourceById,
+  insertResource,
+  updateResourceById,
+  deleteResourceById,
+} from '@/queries/resource';
+import { findAllScopesWithResourceId } from '@/queries/scope';
+import { buildIdGenerator } from '@/utils/id';
+
+import { AuthedRouter } from './types';
+
+const resourceId = buildIdGenerator(21);
+
+export default function resourceRoutes<T extends AuthedRouter>(router: T) {
+  router.get('/resources', koaPagination(), async (ctx, next) => {
+    const { limit, offset } = ctx.pagination;
+
+    const [{ count }, resources] = await Promise.all([
+      findTotalNumberOfResources(),
+      findAllResources(limit, offset),
+    ]);
+
+    // Return totalCount to pagination middleware
+    ctx.pagination.totalCount = count;
+    ctx.body = resources;
+
+    return next();
+  });
+
+  router.post(
+    '/resources',
+    koaGuard({
+      body: Resources.guard
+        .omit({ id: true })
+        .partial()
+        .merge(Resources.guard.pick({ name: true, identifier: true })),
```

Comment:
can we generate a "create" guard if we split return types and update types?

Reply:
yep, I forgot, we already split the update types and return type in the schema db-entries definitions. The `guard` itself is already set as the update type. No need to manually pick the field anymore.  just `Resources. guard.omit({id: retrue})` will do.